rest/rest-smartproxy/src/main/java/org/ow2/proactive_grid_cloud_portal/smartproxy/RestSmartProxyImpl.java

```diff
@@ -312,44 +303,71 @@ public void addEventListener(SchedulerEventListenerExtended listener, boolean my
-        TaskFlowJob tfj = job;
-        for (Task t : tfj.getTasks()) {
-            logger.debug("Pushing files for task " + t.getName());
-            List<String> includes = Lists.newArrayList();
-            List<String> excludes = Lists.newArrayList();
-
-            List<InputSelector> inputFilesList = t.getInputFilesList();
-
-            if (inputFilesList != null) {
-                for (InputSelector is : inputFilesList) {
-                    addfileSelection(is.getInputFiles(), includes, excludes);
-                }
+            if (remotePath == null) {
+                // NOTE: only works for USERSPACE urls
+                throw new IllegalArgumentException("Could not extract remote path using inputSpace=" + inputSpaces +
+                                                   " and userSpace=" + userSpaces);
             }
+            String jname = job.getName();
+            logger.debug("Pushing files for job " + jname + " from " + localInputFolderPath + " to " + remotePath);
+            TaskFlowJob tfj = job;
+            for (Task task : tfj.getTasks()) {
+                uploadInputFilesForTask(localInputFolderPath, remotePath, task);
+            }
+            logger.debug("Finished push operation from " + localInputFolderPath + " to " + remotePath);
 
-            LocalDirSource source = new LocalDirSource(localInputFolderPath);
-            source.setIncludes(includes);
-            source.setExcludes(excludes);
-
-            RemoteDestination dest = new RemoteDestination(USER, remotePath);
-            restDataSpaceClient.upload(source, dest);
+        } catch (Exception error) {
+            throw Throwables.propagate(error);
         }
-        logger.debug("Finished push operation from " + localInputFolderPath + " to " + remotePath);
 
         return true;
     }
 
+    private void uploadInputFilesForTask(String localInputFolderPath, String remotePath, Task task)
+            throws NotConnectedException, PermissionException {
+        logger.debug("Pushing files for task " + task.getName());
+        List<String> includes = Lists.newArrayList();
+        List<String> excludes = Lists.newArrayList();
+
+        List<InputSelector> inputFilesList = task.getInputFilesList();
+
+        if (inputFilesList != null) {
+            for (InputSelector is : inputFilesList) {
+                addfileSelection(is.getInputFiles(), includes, excludes);
+            }
+        }
+
+        LocalDirSource source = new LocalDirSource(localInputFolderPath);
+        source.setIncludes(includes);
+        source.setExcludes(excludes);
+
+        RemoteDestination dest = new RemoteDestination(USER, remotePath);
+        restDataSpaceClient.upload(source, dest);
+    }
+
+    private String extractRelativePath(List<String> userSpaces, List<String> urls) {
+        String remotePath = null;
+        for (int i = 0; i < urls.size(); i++) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)